lib/ansible/modules/packaging/os/satellite_subscription.py

```diff
@@ -0,0 +1,441 @@
+
+        # format json data for request
+        data = {"id": host["id"], "subscriptions": [{"id": available_subscription["id"], "quantity": 1}]}
+
+        # attach the available subscription to this host
+        apicall = "{0}/api/hosts/{1}/subscriptions/add_subscriptions".format(sat_url, host["id"])
+        put(apicall, user, password, data, verify)
+
+        result.append([host["name"], "Assigned: {0}".format(sub["name"])])
+
+    return result, error
+
+
+def subscription_search(sat_url, user, password, org, subscription, module, verify):
+    """Search for subscriptions and return them"""
+
+    matched_subscriptions = {}
+    for subsearch in subscription:
+        # only search for normal subscriptions to avoid returning assigned ones
+        params = {"search": "type=NORMAL and ({0})".format(subsearch)}
+        apicall = "{0}/katello/api/organizations/{1}/subscriptions".format(sat_url, org)
+        matched_subscriptions[subsearch] = request(apicall, user, password, verify, True, params)
+        if matched_subscriptions[subsearch] == []:
+            errors = "Sorry, no subscriptions matched. Please adjust your subscription search"
+            module.fail_json(msg=errors)
+
+    return matched_subscriptions
+
+
+def subscription_remove(sat_url, user, password, host, sub, verify):
+    """Remove a subscription from a host"""
+
+    apicall = "{0}/api/hosts/{1}/subscriptions/remove_subscriptions".format(sat_url, host["id"])
+    data = {"id": host["id"], "subscriptions": [{"id": sub["id"], "quantity": sub["quantity_consumed"]}]}
+    result = put(apicall, user, password, data, verify)
+
+    return result
+
+
+def subscription_configure_autoattach(sat_url, user, password, host, autoheal, verify):
+    """Configure autoattach for a host (autoheal = bool)"""
+    data = {
+        "id": host["id"],
+        "subscription_facet_attributes": {
+            "autoheal": autoheal,
+            "id": host["subscription_facet_attributes"]["id"],
+            "release_version": "",
+            "service_level": "",
+        },
+    }
+    apicall = "{0}/api/hosts/{1}".format(sat_url, host["id"])
+    req = put(apicall, user, password, data, verify)
+
+    return req
+
+
+def subscription_autoattach(sat_url, user, password, host, verify):
+    """Run subscription autoattach for host"""
+
+    data = {"id": host["id"]}
+    apicall = "{0}/api/hosts/{1}/subscriptions/auto_attach".format(sat_url, host["id"])
+    req = put(apicall, user, password, data, verify)
+
+    return req["results"]
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            satellite_hostname=dict(required=True),
+            username=dict(required=True),
+            password=dict(required=True, no_log=True),
+            organization=dict(required=True, type=int),
+            state=dict(
+                default="autoattach",
+                choices=["present", "absent", "autoattach", "disable_autoattach", "vdcguests", "optimize"],
```

Comment:
Please order items in a list alphabetically, unless the order is important for the logic.
```suggestion
                choices=['absent', 'autoattach', 'disable_autoattach', 'optimize', 'present', 'vdcguests'],
```
In fact this could be a single line, also include `type=s'tr'`.